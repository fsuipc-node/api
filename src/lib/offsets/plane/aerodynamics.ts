import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const APU: OffsetList = {
  incidenceAlpha: new Offset({
    value: 0x2ED0,
    name: 'incidenceAlpha',
    description: 'incidence alpha (body AoA) - radians',
    category: OffsetCategory.AERODYNAMICS,
    type: Type.Double,
    permission: 'r',
  }),
};
