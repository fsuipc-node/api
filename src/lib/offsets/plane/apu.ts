import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const APU: OffsetList = {
  APUGeneratorSwitch: new Offset({
    value: 0xB51,
    name: 'APUGeneratorSwitch',
    category: OffsetCategory.APU,
    description: 'APU generator switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  APUGeneratorActive: new Offset({
    value: 0xB52,
    name: 'APUGeneratorActive',
    category: OffsetCategory.APU,
    description: 'APU generator active',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  APUOnFire: new Offset({
    value: 0xB53,
    name: 'APUOnFire',
    category: OffsetCategory.APU,
    description: 'APU generator active',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  APUMaxRPM: new Offset({
    value: 0xB54,
    name: 'APUMaxRPM',
    category: OffsetCategory.APU,
    description: 'APU generator max RPM percent',
    convert: '+({VAL}).toFixed(2)',
    type: Type.Single,
    permission: 'r',
  }),
  APUStarter: new Offset({
    value: 0xB58,
    name: 'APUStarter',
    category: OffsetCategory.APU,
    description: 'APU starter',
    convert: '{VAL} ? 1 : 0',
    type: Type.Single,
    permission: 'rw',
  }),
  APUVoltage: new Offset({
    value: 0xB5C,
    name: 'APUVoltage',
    category: OffsetCategory.APU,
    description: 'APU generator voltage',
    convert: '+({VAL}).toFixed(2)',
    type: Type.Single,
    permission: 'rw',
  }),
};
