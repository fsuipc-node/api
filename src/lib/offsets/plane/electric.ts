import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const electric: OffsetList = {
  masterBatterySwitch: new Offset({
    value: 0x281C,
    name: 'masterBatterySwitch',
    category: OffsetCategory.ELECTRIC,
    description: 'battery master switch',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  totalLoadAmp: new Offset({
    value: 0x2824,
    name: 'totalLoadAmp',
    category: OffsetCategory.ELECTRIC,
    description: 'total load amps',
    type: Type.Double,
    permission: 'r',
  }),
  batteryLoad: new Offset({
    value: 0x282C,
    name: 'batteryLoad',
    category: OffsetCategory.ELECTRIC,
    description: 'battery load',
    type: Type.Double,
    permission: 'r',
  }),
  batteryVoltage: new Offset({
    value: 0x2834,
    name: 'batteryVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'battery voltage',
    type: Type.Double,
    permission: 'r',
  }),
  mainBusVoltage: new Offset({
    value: 0x2840,
    name: 'mainBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'main bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  mainBusAmps: new Offset({
    value: 0x2848,
    name: 'mainBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'main bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  avionicsBusVoltage: new Offset({
    value: 0x2850,
    name: 'avionicsBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'avionics bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  avionicsBusAmps: new Offset({
    value: 0x2858,
    name: 'avionicsBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'avionics bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  hotBatteryBusVoltage: new Offset({
    value: 0x2860,
    name: 'hotBatteryBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'hot battery bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  hotBatteryBusAmps: new Offset({
    value: 0x2868,
    name: 'hotBatteryBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'hot battery bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  batteryBusVoltage: new Offset({
    value: 0x2870,
    name: 'batteryBusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'battery bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  batteryBusAmps: new Offset({
    value: 0x2878,
    name: 'batteryBusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'battery bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator1BusVoltage: new Offset({
    value: 0x2880,
    name: 'generatorAlternator1BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 1 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator1BusAmps: new Offset({
    value: 0x2888,
    name: 'generatorAlternator1BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 1 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator2BusVoltage: new Offset({
    value: 0x2890,
    name: 'generatorAlternator2BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 2 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator2BusAmps: new Offset({
    value: 0x2898,
    name: 'generatorAlternator2BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 2 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator3BusVoltage: new Offset({
    value: 0x28A0,
    name: 'generatorAlternator3BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 3 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator3BusAmps: new Offset({
    value: 0x28A8,
    name: 'generatorAlternator3BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 3 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator4BusVoltage: new Offset({
    value: 0x28B0,
    name: 'generatorAlternator4BusVoltage',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 4 bus voltage',
    type: Type.Double,
    permission: 'r',
  }),
  generatorAlternator4BusAmps: new Offset({
    value: 0x28B8,
    name: 'generatorAlternator4BusAmps',
    category: OffsetCategory.ELECTRIC,
    description: 'generator alternator 4 bus amps',
    type: Type.Double,
    permission: 'r',
  }),
};
