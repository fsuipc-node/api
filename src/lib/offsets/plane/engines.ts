import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

import { engine1 } from './engines/engine1';
import { engine2 } from './engines/engine2';
import { engine3 } from './engines/engine3';
import { engine4 } from './engines/engine4';

export const engines: OffsetList = {
  engineType: new Offset({
    value: 0x609,
    name: 'engineType',
    category: OffsetCategory.ENGINE,
    description: 'engine type',
    convert: 'engineType',
    mapping: true,
    type: Type.Byte,
    permission: 'r',
  }),
  hasMixtureControl: new Offset({
    value: 0x780,
    name: 'hasMixtureControl',
    category: OffsetCategory.ENGINE,
    description: 'has mixture control',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasCarbHeat: new Offset({
    value: 0x784,
    name: 'hasCarbHeat',
    category: OffsetCategory.ENGINE,
    description: 'has carb heat',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  activeEngine: new Offset({
    value: 0x888,
    name: 'activeEngine',
    category: OffsetCategory.ENGINE,
    description: 'active engine pattern',
    type: Type.BitArray,
    length: 1,
    permission: 'rw',
  }),
  ...engine1,
  ...engine2,
  ...engine3,
  ...engine4,
};
