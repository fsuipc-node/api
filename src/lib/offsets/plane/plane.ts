import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const plane: OffsetList = {
  vs0: new Offset({
    value: 0x538,
    name: 'vs0',
    category: OffsetCategory.PLANE,
    description: 'Stall speed full flaps - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vs1: new Offset({
    value: 0x540,
    name: 'vs1',
    category: OffsetCategory.PLANE,
    description: 'Stall speed clean - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vc: new Offset({
    value: 0x548,
    name: 'vc',
    category: OffsetCategory.PLANE,
    description: 'Cruise speed - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vmd: new Offset({
    value: 0x550,
    name: 'vmd',
    category: OffsetCategory.PLANE,
    description: 'Minimum drag speed - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  smokeSystemControl: new Offset({
    value: 0x5D8,
    name: 'smokeSystemControl',
    category: OffsetCategory.PLANE,
    convert: '!!{VAL}',
    description: 'smoke system control on/off',
    type: Type.UInt16,
    permission: 'rw',
  }),
  retractableGear: new Offset({
    value: 0x60C,
    name: 'retractableGear',
    category: OffsetCategory.PLANE,
    description: 'retractable gear (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  retractableLeftFloatExtension: new Offset({
    value: 0x614,
    name: 'retractableLeftFloatExtension',
    category: OffsetCategory.PLANE,
    description: 'retractable left float extension - percent',
    convert: 'Math.floor({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'r',
  }),
  retractableRightFloatExtension: new Offset({
    value: 0x616,
    name: 'retractableRightFloatExtension',
    category: OffsetCategory.PLANE,
    description: 'retractable right float extension - percent',
    convert: 'Math.floor({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'r',
  }),
  hasFlaps: new Offset({
    value: 0x778,
    name: 'hasFlaps',
    category: OffsetCategory.PLANE,
    description: 'has flaps',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasStallHorn: new Offset({
    value: 0x77C,
    name: 'hasStallHorn',
    category: OffsetCategory.PLANE,
    description: 'has stall horn',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasMixtureEngine: new Offset({
    value: 0x780,
    name: 'hasMixtureEngine',
    category: OffsetCategory.PLANE,
    description: 'has mixture engine',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasCarbHeat: new Offset({
    value: 0x784,
    name: 'hasCarbHeat',
    category: OffsetCategory.PLANE,
    description: 'has carburator heater',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasSpoilers: new Offset({
    value: 0x78C,
    name: 'hasSpoilers',
    category: OffsetCategory.PLANE,
    description: 'has spoilers',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  isTailDragger: new Offset({
    value: 0x790,
    name: 'isTailDragger',
    category: OffsetCategory.PLANE,
    description: 'has spoilers',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasStrobes: new Offset({
    value: 0x794,
    name: 'hasStrobes',
    category: OffsetCategory.PLANE,
    description: 'has strobes',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasToeBrakes: new Offset({
    value: 0x79C,
    name: 'hasToeBrakes',
    category: OffsetCategory.PLANE,
    description: 'has toe brakes',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  maxMach: new Offset({
    value: 0xB0C,
    name: 'maxMach',
    category: OffsetCategory.PLANE,
    description: 'mach maximum operating - not FS2004',
    convert: '+({VAL} / 20438).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
  fuelSelectorsCount: new Offset({
    value: 0x123E,
    name: 'fuelSelectorsCount',
    category: OffsetCategory.PLANE,
    description: 'number of fuel selectors for the aircraft',
    type: Type.Byte,
    permission: 'r',
  }),
  aircraftEmptyWeight: new Offset({
    value: 0x1330,
    name: 'aircraftEmptyWeight',
    category: OffsetCategory.PLANE,
    description: 'aicraft empty weight',
    convert: '+({VAL} / 256).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
  aircraftMaxGrossWeight: new Offset({
    value: 0x1334,
    name: 'aircraftMaxGrossWeight',
    category: OffsetCategory.PLANE,
    description: 'aicraft max gross weight',
    convert: '+({VAL} / 256).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
};
