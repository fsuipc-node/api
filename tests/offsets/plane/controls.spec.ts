import { controls as offsets } from '@offsets/plane/controls';

describe('offset - plane/controls', () => {
  const offsetsTestCases = [
    { name: 'autoRudder', value: 0, expectedResult: false },
    { name: 'autoRudder', value: 1, expectedResult: true },
    { name: 'rotorClutchSwitch', value: 0, expectedResult: false },
    { name: 'rotorClutchSwitch', value: 1, expectedResult: true },
  ];

  describe('offsets list', () => {
    it('should have required properties', () => {
      expect(offsets).toMatchSnapshot();
    });
  });

  offsetsTestCases.forEach(testedOffset => {
    describe(testedOffset.name, () => {
      it('should convert data properly', () => {
        const convertExpression = offsets[testedOffset.name].convert.replace(new RegExp(/{VAL}/g), testedOffset.value.toString());

        // tslint:disable-next-line:no-eval
        expect(eval(convertExpression)).toEqual(testedOffset.expectedResult);
      });
    });
  });
});
