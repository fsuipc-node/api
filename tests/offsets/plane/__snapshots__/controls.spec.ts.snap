// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offset - plane/controls offsets list should have required properties 1`] = `
Object {
  "aileronDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "aileron deflection - radians +ve right -ve left",
    "length": undefined,
    "mapping": undefined,
    "name": "aileronDeflection",
    "permission": "r",
    "type": 8,
    "value": 11944,
  },
  "aileronIndicator": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "aileron indicator",
    "length": undefined,
    "mapping": undefined,
    "name": "aileronIndicator",
    "permission": "r",
    "type": 2,
    "value": 3000,
  },
  "aileronPositionControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "aileron position control - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "aileronPositionControl",
    "permission": "rw",
    "type": 2,
    "value": 2998,
  },
  "aileronTrimDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "aileron trim deflection - radians +ve right -ve left",
    "length": undefined,
    "mapping": undefined,
    "name": "aileronTrimDeflection",
    "permission": "rw",
    "type": 8,
    "value": 11952,
  },
  "aileronTrimPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "aileron trim position - percent - can be negative",
    "length": undefined,
    "mapping": undefined,
    "name": "aileronTrimPosition",
    "permission": "rw",
    "type": 2,
    "value": 3074,
  },
  "autoBrakeSwitch": Offset {
    "category": "controls",
    "convert": "autoBrake",
    "description": "auto brake switch",
    "length": undefined,
    "mapping": true,
    "name": "autoBrakeSwitch",
    "permission": "rw",
    "type": 1,
    "value": 12160,
  },
  "autoRudder": Offset {
    "category": "controls",
    "convert": "!!{VAL}",
    "description": "auto coordination",
    "length": undefined,
    "mapping": undefined,
    "name": "autoRudder",
    "permission": "rw",
    "type": 2,
    "value": 632,
  },
  "brakingIndicator": Offset {
    "category": "controls",
    "convert": "appliedBrakes",
    "description": "braking indicator",
    "length": undefined,
    "mapping": true,
    "name": "brakingIndicator",
    "permission": "r",
    "type": 2,
    "value": 3018,
  },
  "canopyOpen": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "canopy open",
    "length": undefined,
    "mapping": undefined,
    "name": "canopyOpen",
    "permission": "r",
    "type": 8,
    "value": 10832,
  },
  "elevatorDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevator deflection - radians +ve up -ve down",
    "length": undefined,
    "mapping": undefined,
    "name": "elevatorDeflection",
    "permission": "r",
    "type": 8,
    "value": 11928,
  },
  "elevatorIndicator": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevator indicator",
    "length": undefined,
    "mapping": undefined,
    "name": "elevatorIndicator",
    "permission": "r",
    "type": 2,
    "value": 2996,
  },
  "elevatorPositionControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "elevator position control - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "elevatorPositionControl",
    "permission": "rw",
    "type": 2,
    "value": 2994,
  },
  "elevatorTrimDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevator trim deflection - radians +ve up -ve down",
    "length": undefined,
    "mapping": undefined,
    "name": "elevatorTrimDeflection",
    "permission": "rw",
    "type": 8,
    "value": 11936,
  },
  "elevatorTrimIndicator": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevator trim indicator",
    "length": undefined,
    "mapping": undefined,
    "name": "elevatorTrimIndicator",
    "permission": "r",
    "type": 2,
    "value": 3010,
  },
  "elevatorTrimPositionControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "elevator trim position control - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "elevatorTrimPositionControl",
    "permission": "rw",
    "type": 2,
    "value": 3008,
  },
  "elevon1Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 1 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon1Deflection",
    "permission": "r",
    "type": 8,
    "value": 10752,
  },
  "elevon2Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 2 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon2Deflection",
    "permission": "r",
    "type": 8,
    "value": 10760,
  },
  "elevon3Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 3 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon3Deflection",
    "permission": "r",
    "type": 8,
    "value": 10768,
  },
  "elevon4Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 4 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon4Deflection",
    "permission": "r",
    "type": 8,
    "value": 10776,
  },
  "elevon5Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 5 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon5Deflection",
    "permission": "r",
    "type": 8,
    "value": 10784,
  },
  "elevon6Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 6 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon6Deflection",
    "permission": "r",
    "type": 8,
    "value": 10792,
  },
  "elevon7Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 7 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon7Deflection",
    "permission": "r",
    "type": 8,
    "value": 10800,
  },
  "elevon8Deflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "elevon 8 deflection",
    "length": undefined,
    "mapping": undefined,
    "name": "elevon8Deflection",
    "permission": "r",
    "type": 8,
    "value": 10800,
  },
  "flapsControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "flaps deployment - percent - for steps, depends on 0x3BFA",
    "length": undefined,
    "mapping": undefined,
    "name": "flapsControl",
    "permission": "rw",
    "type": 3,
    "value": 3036,
  },
  "flapsHandleIndex": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "0 = full up",
    "length": undefined,
    "mapping": undefined,
    "name": "flapsHandleIndex",
    "permission": "rw",
    "type": 0,
    "value": 3068,
  },
  "flapsLeftPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "left flaps position indicator - percent - to get angle, depends on max",
    "length": undefined,
    "mapping": undefined,
    "name": "flapsLeftPosition",
    "permission": "r",
    "type": 3,
    "value": 3040,
  },
  "flapsRightPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "right flaps position indicator - percent - to get angle, depends on max",
    "length": undefined,
    "mapping": undefined,
    "name": "flapsRightPosition",
    "permission": "r",
    "type": 3,
    "value": 3040,
  },
  "flightDirectorActive": Offset {
    "category": "controls",
    "convert": "!!{VAL}",
    "description": "panel auto-feather arm switch",
    "length": undefined,
    "mapping": undefined,
    "name": "flightDirectorActive",
    "permission": "rw",
    "type": 6,
    "value": 12000,
  },
  "flightDirectorBank": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "flight director bank - degrees",
    "length": undefined,
    "mapping": undefined,
    "name": "flightDirectorBank",
    "permission": "r",
    "type": 8,
    "value": 12016,
  },
  "flightDirectorPitch": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "flight director pitch - degrees",
    "length": undefined,
    "mapping": undefined,
    "name": "flightDirectorPitch",
    "permission": "r",
    "type": 8,
    "value": 12008,
  },
  "foldingWingLeft": Offset {
    "category": "controls",
    "convert": "doublePercentRound",
    "description": "folding right left - percent",
    "length": undefined,
    "mapping": true,
    "name": "foldingWingLeft",
    "permission": "r",
    "type": 8,
    "value": 10824,
  },
  "foldingWingRight": Offset {
    "category": "controls",
    "convert": "doublePercentRound",
    "description": "folding right right - percent",
    "length": undefined,
    "mapping": true,
    "name": "foldingWingRight",
    "permission": "r",
    "type": 8,
    "value": 10832,
  },
  "gearControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "gear deploy - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "gearControl",
    "permission": "rw",
    "type": 3,
    "value": 3048,
  },
  "gearLeftPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "nose gear deploy - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "gearLeftPosition",
    "permission": "r",
    "type": 3,
    "value": 3060,
  },
  "gearNosePosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "nose gear deploy - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "gearNosePosition",
    "permission": "r",
    "type": 3,
    "value": 3052,
  },
  "gearPosTail": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "gear pos tail",
    "length": undefined,
    "mapping": undefined,
    "name": "gearPosTail",
    "permission": "r",
    "type": 8,
    "value": 12088,
  },
  "gearRightPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "nose gear deploy - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "gearRightPosition",
    "permission": "r",
    "type": 3,
    "value": 3056,
  },
  "leftAileronDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "left aileron deflection - radians",
    "length": undefined,
    "mapping": undefined,
    "name": "leftAileronDeflection",
    "permission": "r",
    "type": 8,
    "value": 944,
  },
  "leftBrakeApplication": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "left brake application pressure - percent - will match 100% if parking brake on",
    "length": undefined,
    "mapping": undefined,
    "name": "leftBrakeApplication",
    "permission": "r",
    "type": 2,
    "value": 3012,
  },
  "leftToeBrakeControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 2)",
    "description": "left toe brake application - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "leftToeBrakeControl",
    "permission": "rw",
    "type": 0,
    "value": 3073,
  },
  "parkingBrake": Offset {
    "category": "controls",
    "convert": "typeof {VAL} === \`boolean\` ? {VAL} ? 32767 : 0 : {VAL} >= 32767",
    "description": "parking brake active",
    "length": undefined,
    "mapping": undefined,
    "name": "parkingBrake",
    "permission": "r",
    "type": 2,
    "value": 3016,
  },
  "propellerPitchControl": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "0=fixed, 1=auto, 2=manual",
    "length": undefined,
    "mapping": undefined,
    "name": "propellerPitchControl",
    "permission": "rw",
    "type": 5,
    "value": 2800,
  },
  "rightAileronDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "right aileron deflection - radians",
    "length": undefined,
    "mapping": undefined,
    "name": "rightAileronDeflection",
    "permission": "r",
    "type": 8,
    "value": 952,
  },
  "rightBrakeApplication": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "right brake application pressure - percent - will match 100% if parking brake on",
    "length": undefined,
    "mapping": undefined,
    "name": "rightBrakeApplication",
    "permission": "r",
    "type": 2,
    "value": 3014,
  },
  "rightToeBrakeControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 2)",
    "description": "right toe brake application - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "rightToeBrakeControl",
    "permission": "rw",
    "type": 0,
    "value": 3072,
  },
  "rotorClutchSwitch": Offset {
    "category": "controls",
    "convert": "!!{VAL}",
    "description": "rotor clutch switch",
    "length": undefined,
    "mapping": undefined,
    "name": "rotorClutchSwitch",
    "permission": "rw",
    "type": 0,
    "value": 2185,
  },
  "rudderCalibratedvalue": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "rudder calibrated value - percent - can be negative",
    "length": undefined,
    "mapping": undefined,
    "name": "rudderCalibratedvalue",
    "permission": "r",
    "type": 2,
    "value": 3082,
  },
  "rudderDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "rudder deflection - radians +ve right -ve left",
    "length": undefined,
    "mapping": undefined,
    "name": "rudderDeflection",
    "permission": "r",
    "type": 8,
    "value": 11960,
  },
  "rudderIndicator": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "rudder indicator",
    "length": undefined,
    "mapping": undefined,
    "name": "rudderIndicator",
    "permission": "r",
    "type": 2,
    "value": 3004,
  },
  "rudderPositionControl": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "rudder position control - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "rudderPositionControl",
    "permission": "rw",
    "type": 2,
    "value": 3002,
  },
  "rudderTrimDeflection": Offset {
    "category": "controls",
    "convert": undefined,
    "description": "rudder trim deflection - radians +ve right -ve left",
    "length": undefined,
    "mapping": undefined,
    "name": "rudderTrimDeflection",
    "permission": "rw",
    "type": 8,
    "value": 11968,
  },
  "rudderTrimPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "rudder trim position - percent - can be negative",
    "length": undefined,
    "mapping": undefined,
    "name": "rudderTrimPosition",
    "permission": "rw",
    "type": 2,
    "value": 3076,
  },
  "spoilerLeftPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "left spoiler position - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "spoilerLeftPosition",
    "permission": "rw",
    "type": 3,
    "value": 3028,
  },
  "spoilerRightPosition": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "right spoiler position - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "spoilerRightPosition",
    "permission": "rw",
    "type": 3,
    "value": 3032,
  },
  "spoilersArm": Offset {
    "category": "controls",
    "convert": "!!{VAL}",
    "description": "spoilers arm indicator",
    "length": undefined,
    "mapping": undefined,
    "name": "spoilersArm",
    "permission": "rw",
    "type": 3,
    "value": 3020,
  },
  "spoilersControl": Offset {
    "category": "controls",
    "convert": "spoilersControl",
    "description": "spoilers control - percent",
    "length": undefined,
    "mapping": true,
    "name": "spoilersControl",
    "permission": "rw",
    "type": 3,
    "value": 3024,
  },
  "steeringTillerCalibratedValue": Offset {
    "category": "controls",
    "convert": "Math.round({VAL} / 16383 * 100)",
    "description": "steering tiller calibrated value - percent - can be negative",
    "length": undefined,
    "mapping": undefined,
    "name": "steeringTillerCalibratedValue",
    "permission": "r",
    "type": 2,
    "value": 3080,
  },
  "tailWheelLocked": Offset {
    "category": "controls",
    "convert": "!!{VAL}",
    "description": "tail wheel locked",
    "length": undefined,
    "mapping": undefined,
    "name": "tailWheelLocked",
    "permission": "rw",
    "type": 6,
    "value": 10896,
  },
}
`;
